src/variables/components/TypeAheadVariableDropdown.tsx

```diff
@@ -0,0 +1,308 @@
+    }
+  }
+
+  filterVals = needle => {
+    const {values} = this.props
+
+    if (!needle) {
+      this.setState({shownValues: values, typedValue: needle})
+    } else {
+      const result = values.filter(val =>
+        val.toLowerCase().includes(needle.toLowerCase())
+      )
+      this.setState({
+        shownValues: result,
+        typedValue: needle,
+        menuOpen: MenuStatus.Open,
+      })
+    }
+  }
+
+  maybeSelectNextItem = e => {
+    const {shownValues, selectIndex} = this.state
+
+    let newIndex = -1
+
+    if (e.keyCode === 40) {
+      // down arrow
+      newIndex = selectIndex + 1
+    } else if (e.keyCode === 38) {
+      // up arrow
+      newIndex = selectIndex - 1
+    }
+
+    const numItems = shownValues.length
+
+    if (numItems && newIndex >= 0 && newIndex < numItems) {
+      this.handleSelect(shownValues[newIndex], newIndex)
+      return
+    }
+
+    if (e.keyCode === 13) {
+      // return/enter key
+      // lose focus, reset the selectIndex to -1, & close the menu:
+      e.target.blur()
+
+      // the person could have been typing and pressed return, need to reset the value
+      // back to the 'real value'
+      const newState = {
+        menuOpen: MenuStatus.Closed,
+        selectIndex: -1,
+        ...this.getRealValue(),
+      }
+      this.setState(newState)
+    }
+  }
+
+  // if the entire item loses focus, then
+  // the input should show the actual selected item, not what the user typed in;
+  // only want to show valid values when the component is not actively being used
+  onClickAwayHere = () => {
+    this.setState(this.getRealValue())
+  }
+
+  getRealValue = () => {
+    const {actualVal, typedValue} = this.state
+    const {selectedValue} = this.props
+
+    if (actualVal || selectedValue) {
+      const realValue = actualVal ?? selectedValue
+
+      if (typedValue !== realValue) {
+        return {typedValue: realValue}
+      }
+    }
+  }
+
+  render() {
+    const {selectedValue, values, name} = this.props
+    const {typedValue, shownValues, menuOpen} = this.state
+
+    const dropdownStatus =
+      values.length === 0 ? ComponentStatus.Disabled : ComponentStatus.Default
+
+    const placeHolderText = this.getPlaceHolderText('Select a Value')
+
+    const widthStyle = this.getWidth(placeHolderText)
+
+    return (
+      <Dropdown
+        style={{width: '140px'}}
+        className="variable-dropdown--dropdown"
+        testID={this.props.testID || `variable-dropdown--${name}`}
+        onClickAway={this.onClickAwayHere}
+        menuOpen={menuOpen}
+        button={(active, onClick) => (
+          <Dropdown.Button
+            active={active}
+            onClick={onClick}
+            testID="variable-dropdown--button"
+            status={dropdownStatus}
+          >
+            <Input
+              style={widthStyle}
+              placeholder={placeHolderText}
+              onChange={e => this.filterVals(e.target.value)}
```

Comment:
it's typical convention to just pass in the handler, and let it handle the event being passed in. so by making both handlers take in an event, it implicitly communicates it's an event handler. so `filterVals` argument should change from `needle` to `event`. and this call would look like:

```onChange={this.filterVals}```